### ISSUE TITLE
Track changes of Dokumente_zur_Entstehungsgeschichte.xls

### ISSUE THREAD

**Issue by gerritbruening** [faustedition/xml#18]
@thvitt , we just had to watchout for possible corruption in file `Dokumente zur Entstehungsgeschichte.xls` from https://faustedition.uni-wuerzburg.de/wiki/index.php/17._M%C3%A4rz_2017.
Would it be feasible to systematically track changes of this file? In the Wiki you only see that there are different versions. You cannot see the differences.
Idea:
* load the version from the wiki
* convert it to, say, csv or whatever is suitable
* make a readible diff like in git

**Comment by thvitt** [comment]
Theoretically, yes.

Practically, it would be helpful to know about ① the kind of corruption and ② the revisions (by date) between which the corruption probably occured. People seem to be constantly adding columns, making the (line-oriented) diffs light up brightly.

**Comment by gerritbruening** [comment]
Version from 9. Feb. 2017 has reordered rows, what should have been avoided. 
DP has undone this this reordering earlier today (13:29, 6. Mär. 2017). Hence, the most interesting comparison would be between 17. Okt. 2016 and the current version.

**Comment by gerritbruening** [comment]
Sorry, the interesting comparison would of course be between 17. Okt. 2016 and **13:29**, 6. Mär. 2017. Because after this a column was added.

**Comment by thvitt** [comment]
```diff
--- 20170209171813!Dokumente_zur_Entstehungsgeschichte.tsv	2017-03-06 15:07:39.702617446 +0100
+++ 20170306133404!Dokumente_zur_Entstehungsgeschichte.tsv	2017-03-06 15:07:40.262614786 +0100
@@ -881,7 +881,7 @@
 								II-980	Johann Friedrich von Cotta	36466	Brief	letter		Johann Wolfgang Goethe	n.v.				03.03.1827									Kuhn II, 196f	QuZ II, 434				quz_2_0434-0435						
 	1465								Johann Wolfgang Goethe	n.v.	Tagebuch	diary							09.03.1827								WA III, 11, 30	WA III, 11, 30							30	9			
 								II-991	Wilhelm Reichel	40581	Brief	letter		Johann Wolfgang Goethe	n.v.				11.03.1827						NFG/GSA Goethe-Akten 315, 111			QuZ II, 437					quz_2_0436, quz_2_0437						
-	1466		510						Johann Wolfgang Goethe	n.v.	Brief			Karl Friedrich von Reinhard	40609				12.03.1827								WA IV, 42, 81	Reinhard 1850, S. 284											
+	1466		510						Johann Wolfgang Goethe	n.v.	Brief			Karl Friedrich von Reinhard	40609				30.03.1827						Bei Pniower und Gräf falsch datiert		WA IV, 42, 111	Reinhard 1850, S. 284											
 	1466a								Johann Wolfgang Goethe	n.v.	Tagebuch	diary							12.03.1827					s. Gräf II/1, Nr. 251			WA III, 11, 32	WA III, 11, 32							32	12			
 	1467		513						Johann Wolfgang Goethe	n.v.	Text	1. article 2. draft	Entwurf zu einer Rezension von Hinrichs' 'Das Wesend er antiken Tragödie'						15.03.1827	00.04.1827		Ende März oder Anfang April 1827; Jahr ergänzt					WA I, 15 (2), 214f.	WA I, 15 (2), 214f.				nicht in Zeno			214	2	215		
 	1468		511					II-999	Johann Wolfgang Goethe	n.v.	Brief			Wilhelm Reichel	40581				18.03.1827								WA IV, 42, 88	WA I, 15 (2), 127							126	Z. 15		Z. ?	
@@ -1437,7 +1437,7 @@
 	1938		896					II-2023	Johann Wolfgang Goethe	n.v.	Tagebuch	diary							15.07.1831						Vgl. Z. 29-35		WA III, 13, 109	WA III, 13, 109							109	15			
 	1939		897					II-2024	Johann Wolfgang Goethe	n.v.	Tagebuch	diary							18.07.1831						vgl. Nr. 1483		WA III, 13, 110	WA III, 13, 110							110	18			
 	1940		898					II-2025	Johann Wolfgang Goethe	n.v.	Tagebuch	diary							19.07.1831								WA III, 13, 110	WA III, 13, 110							110	19			
-	1941		900					II-2027	Johann Wolfgang Goethe	n.v.	Brief			Johann Heinrich Meyer	44570				20.07.1831						Wegen 586, 1f vgl. 607, 14-16.; wegen 586, 14-16 vgl. 589, 9f. 590, 12f., ebenso das Folgende		WA IV, 49, 11	KuA VI/3, S. 618f.											
+	1941		900					II-2027	Johann Wolfgang Goethe	n.v.	Brief			Johann Heinrich Meyer	44570				20.07.1831						Wegen 586, 1f vgl. 607, 14-16.; wegen 586, 14-16 vgl. 589, 9f. 590, 12f., ebenso das Folgende/ in wa Konzept, umfangreicher in KuA		WA IV, 49, 11	KuA VI/3, S. 618f.											
 	1942		899					II-2026	Johann Wolfgang Goethe	n.v.	Tagebuch	diary							20.07.1831								WA III, 13, 111	WA III, 13, 111							111	20			
 	1943		901					II-2028	Johann Wolfgang Goethe	n.v.	Tagebuch	diary							21.07.1831								WA III, 13, 112	WA III, 13, 112							112	21			
 	1944		903						Johann Wolfgang Goethe	n.v.	Brief			Sulpiz Boisserée	43183				22.07.1831								WA IV, 49, 14	Boisserée 1862, Bd. 2, S. 571									
```

**Comment by gerritbruening** [comment]
Great, thanks!
What is `@@ -881,7 +881,7 @@` for?

**Comment by thvitt** [comment]
[line numbers in the original files](https://www.gnu.org/software/diffutils/manual/html_node/Detailed-Unified.html#Detailed-Unified).

**Comment by gerritbruening** [comment]
That's what I thought, but what does `,7` mean?

**Comment by thvitt** [comment]
> […] Otherwise its line numbers look like ‘start,count’.

**Comment by gerritbruening** [comment]
I think I don't understand the concepts 'start' and 'count' ...

**Comment by thvitt** [comment]
Have a look at [the first hunk of generate-all.xpl here](https://github.com/faustedition/faust-gen-html/commit/e7462950b928b405bdd1afd3680707ab9640cf78). If you look at the first two columns in Github's representation of that patch, you see that they represent the line numbers of generate-all.xpl before (left) and after (right) the change (the addition of the empty line and the import statement).

Now, that hunk's header says `@@ -27,6 +27,8 @@`. The first pair of numbers (`27,6`) says that the following diff represents the 6 lines from line number 27 (`<p:import href="generate-metadata-js.xpl"/>`) of generate-all.xpl before the patch has been applied. 

The second pair of numbers (`27,8`) says the diff represents the 8 lines starting at line 27 _after_ the patch has been applied – 8 lines because the patch adds 2 lines to the original 6. 

In its default setting, `diff -u` adds three unchanged lines of context before and after each changed line, so a single modified line is represented by a 7-line-long patch (3 unchanged lines, changed line (represented in the version before `-` and after `+` the modification), 3 unchanged lines). Thus, in the first modification of the excel file,  `@@ -881,7 +881,7 @@`, line 881 is the one with II-980, which is unmodified, the actually modified line is the one with Gräf-Nr. 1466, which is line no. 884 in the CSV files that I've compared.

tl;dr: Only look at the coloured lines.

(Now I could have used the last 15 minutes for writing code that actually does something with your messy excel tables like colouring the lines by 'Lfd. Nr. megakrassneu' …)

**Comment by gerritbruening** [comment]
> tl;dr:
Was a pleasure to read it, now I see!

I think we no in fact have "allerneuestens" in the xls but I don't know exactly ...

**Comment by gerritbruening** [comment]
@sandrakrause , we observed that a version of `Dokumente_zur_Entstehungsgeschichte.xls` was uploaded to https://faustedition.uni-wuerzburg.de/wiki/index.php/Datei:Dokumente_zur_Entstehungsgeschichte.xls that had been resorted by column “WA Druckort” (if I remember correctly). Resorting is always critical because the whole spreadsheet is changed so that the few ‘real’ changes of single fields are obscured. Moreover, if something goes wrong in the sorting (for example, sorting without “Markierung erweitern”), the whole file is full of errors.
After JK did some additional changes (February 14), DP restored the original sorting by column “lfd. Nr. (neuestens)” yesterday.
After that, TV ran a comparison of the last version before resorting and the version with the resorting undone. Now the “real” changes became visible, see the [visualisation above](https://github.com/faustedition/xml/issues/18#issuecomment-284410900). Could you please check whether you see all the changes that you intended to do? DP mentioned that he advised you to delete a row, but I do not see this change in the visualisation. In you “Kommentar” you only report about changes made by JK.
As you see, the current version has some significant changes made by DP.
However, when JK tried to open this version, the browser did not open this version but the previous one.
As you see, working with Excel and the Wiki is messy. So when you edit the spreadsheet in the future:
* Make sure that your browser has in fact opened the current version (are the changes mentioned in “Kommentar” visible in the file you see?)
* You may resort the table, but preferably only to work on the XML files, not to edit and upload the table.
* When you edit the table, edit a version that is not resorted.
* If you find this unpractical because it is necessary to resort the table before making the changes, contact DP.
* If you edit and resort the table, resort the table by column “lfd. Nr. (allerneuestens)” afterwards before uploading the new version in the wiki. If I remember correctly, this somehow did not work when we both tried it on your device. Please try to find out how you can do this on your device. Maybe it works with the new column “lfd. Nr. (allerneuestens)”?
* When you add or remove columns, it is best to do that in an isolated step that is present in the wiki version history. This version should differ from the previous one only by the addition or removal of columns.
* As soon as you think that changes ought to be tracked, contact TV and ask him to compare these version. By this you can make sure that the changes you intended to make are in fact present in the new version and that there are no unintended changes.

**Comment by sandrakrause** [comment]
In fact, I did not change anything in the uploaded spreadsheet yet.
The only version I uploaded was the one with the changes of JK, which we were not able to open until I reuploaded it. The reason I report only changes made by JK is that I did not make/upload any changes.
DP told me to mark a row as "Entfällt" but I did not upload it yet.  DP and me changed this row in the spreadsheet I was working with (which was sorted by another column than "Lfd Nr."), so I could not directly upload it, but wanted to repeat the changes to the original unresorted table, but had not enough time that day. In order to prevent mistakes stated above, I did not want to rush things and planned on uploading it this week.
I'm sorry that this behavior led to misunderstandings.

**Comment by gerritbruening** [comment]
Great, thanks for your clarification!
I did not know that you are in fact perfectly aware of the problem of resorting (because I did not exactly remember whether we discussed this intensively). But I was aware that it was not you that did the resorting of the uploaded version :-)
There is no reason to apologize because there is no individual mistake. The error is in the Excel-wiki way of working. 
I did not know that there are files that cannot be opened but uploaded---one more reason to move away from the current practice …
